File: stone_processor/src/main/java/com/github/klee0kai/stone/helpers/wrap/WrapType.java
package com.github.klee0kai.stone.helpers.wrap;

import com.squareup.javapoet.TypeName;

public class WrapType {

    public TypeName typeName;

    public boolean isNoCachingWrapper = false;

    public FormatSimple unwrap;

    public FormatSimple wrap;

    public FormatInList inListFormat;

    public boolean isList() {
        return inListFormat != null;
    }
}

File: stone_ksp/src/commonMain/kotlin/com/github/klee0kai/thekey/stone/ksp/helpers/wrap/WrapType.kt
package com.github.klee0kai.thekey.stone.ksp.helpers.wrap

import com.squareup.kotlinpoet.TypeName

data class WrapType(
    val typeName: TypeName,
    val unwrap: UnwrapFun,
    val wrap: WrapFun,
    val isNoCachingWrapper: Boolean = true,
    val inListFormat: FormatInList? = null,
) {
    val isList: Boolean get() = inListFormat != null
}

